### PR TITLE
Add Python 3 support

### LINKED ISSUE #175: Add support for Python 3
看了一下https://pypi.python.org的轮子，发现只有python2.7，作者可否支持python3?非常感谢提供这么好的工具

### PR DESCRIPTION
**Pull request by oraoto**
fix #175 

* Add Python 3 support
* Make test and build scripts compatible with Python 2 & 3

### CODE REVIEW

**Comment by varunarora** [comment]
Hey @oraoto, love the contributions! Thanks for jumping on some of our most important issues so quickly.

@Superjomn You got this? I think your the best person to do justice to this PR.

**Comment by jetfuel** [comment]
Great Work! @oraoto Thank you so much. 
@Superjomn @jacquesqiao Please take a look at your best convenience.

**Comment by jetfuel** [comment]
There are some warnings in the pybind. We can update the code to migrate the "__init__" in pybind.cc to 

```
# LogReader 
      .def(py::init( [](const std::string& dir) {

        return new vs::LogReader(dir);

      }))

# LogWriter
     .def(py::init( [](const std::string& dir, int sync_cycle) {
        return new vs::LogWriter(dir, sync_cycle);
      }))
```

**Comment by jetfuel** [comment]
Files under /demo should update afterward. We can create a separated issue to address that later.

File: visualdl/python/test_storage.py [hunk 1]
```diff
@@ -41,9 +42,9 @@ def test_image(self):
         num_samples = 100
         shape = [10, 10, 3]
 
-        for pass_ in xrange(num_passes):
+        for pass_ in range(num_passes):
```
Comment (Superjomn) [inline visualdl/python/test_storage.py]: why these `xrange` changed to `range`?

File: visualdl/storage/CMakeLists.txt [hunk 2]
```diff
@@ -14,3 +14,5 @@ add_dependencies(entry storage_proto im)
 add_dependencies(record storage_proto entry)
 add_dependencies(tablet storage_proto)
 add_dependencies(storage storage_proto record binary_record tablet entry)
+
+target_link_libraries(storage storage_proto record binary_record tablet entry)
```
Comment (Superjomn) [inline visualdl/storage/CMakeLists.txt]: the pre-commit should be executed, that will help to reformat the code style and some delete such empty lines.

run pre-commit by `pre-commit run -a`, need to install first.

Comment (oraoto) [reply visualdl/storage/CMakeLists.txt]: Sorry for not reading the develop guide carefully. Coding style fixed.

**Review by jetfuel — APPROVED** [review]
LGTM

**Pull request merged**